src/views/CashlinkReceive.vue

```diff
@@ -284,31 +290,37 @@ class CashlinkReceive extends Vue {
         window.location.href = Config.redirectTarget;
     }
 
-    private get isCashlinkStateKnown(): boolean {
-        return this.cashlink!.state !== CashlinkState.UNKNOWN;
-    }
-
     private get canCashlinkBeClaimed(): boolean {
         return this.cashlink!.state === CashlinkState.UNCLAIMED;
     }
 
     private get buttonText(): string {
-        if (!this.isCashlinkStateKnown) return this.$t('Checking status') as string;
-        else if (this.canCashlinkBeClaimed) return this.$t('Claim Cashlink') as string;
-        else if (this.cashlink!.state === CashlinkState.UNCHARGED) return this.$t('Cashlink not funded') as string;
-        else if (this.cashlink!.state === CashlinkState.CHARGING) return this.$t('Cashlink funding') as string;
-        else {
-            if (!this.isClaiming) {
+        if (this.cashlink!.state !== CashlinkState.CLAIMED && this.statusState === StatusScreen.State.WARNING) {
+            // Reset warning screen if cashlink state reverted from CLAIMING state, e.g. by being refunded.
+            this.statusState = false;
+        }
```

Comment:
The condition checks for `CLAIMED` state, but the comment talks about `CLAIMING` state. Mistake or on purpose and needs clarification?